missingdrone/missingdrone.js

```diff
@@ -0,0 +1,98 @@
+
+  function findUniqueDeliveryId(deliveryIds) {
+
+  const idsToOccurrences = new Map();
+
+  deliveryIds.forEach(deliveryId => {
+    if (idsToOccurrences.has(deliveryId)) {
+      const newCount = idsToOccurrences.get(deliveryId) + 1;
+      idsToOccurrences.set(deliveryId, newCount);
+    } else {
+      idsToOccurrences.set(deliveryId, 1);
+    }
+  });
+
+  return [...idsToOccurrences.keys()].find(id => idsToOccurrences.get(id) === 1);
+}
+
+// You could use an object instead of a Map here if you wanted. The drawback is that object keys must be strings or symbols, so our keys would be coerced to strings (1 would become '1').
+
+// Alright, we got our runtime down to O(n)O(n). That's probably the best runtime we can get—to find our unique integer we’ll definitely have to look at every integer in the worst case.
+
+// But now we've added O(n)O(n) space, for our object. Can we bring that down?
+
+// Well, we could use booleans ↴ as our values, instead of integers. If we see an integer, we'll add it as a key in our object with a boolean value of true. If we see it again, we'll change its value to false. At the end, our non-repeated order ID will be the one integer with a value of true.
+
+// How much space does this save us? Depends how our language stores booleans vs integers. Often booleans take up just as much space as integers.
+
+// And even if each boolean were just 1 bit, that'd still be O(n)O(n) space overall.
+
+// So using booleans probably doesn't save us much space here. Any other ideas?
+
+// Let’s zoom out and think about what we’re working with. The only data we have is integers. How are integers stored?
+
+// Our machine stores integers as binary numbers ↴ using bits. ↴ What if we thought about this problem on the level of individual bits?
+
+// Let's think about the bitwise operations AND, ↴ OR, ↴ XOR, ↴ NOT ↴ and bit shifts. ↴
+
+// Is one of those operations helpful for finding our unique integer?
+
+// We’re seeing every integer twice, except one. Is there a bitwise operation that would let the second occurrence of an integer cancel out the first?
+
+// If so, we could start with a variable uniqueDeliveryId set to 00 and run some bitwise operation with that variable and each number in our array. If duplicate integers cancel each other out, then we’d only be left with the unique integer at the end!
+
+// Which bitwise operation would let us do that?
+
+// Solution
+// We XOR ↴ all the integers in the array. We start with a variable uniqueDeliveryId set to 00. Every time we XOR with a new ID, it will change the bits. When we XOR with the same ID again, it will cancel out the earlier change.
+
+// In the end, we'll be left with the ID that appeared once!
+
+  function findUniqueDeliveryId(deliveryIds) {
+
+  var uniqueDeliveryId = 0;
+
+  deliveryIds.forEach(deliveryId => {
+    uniqueDeliveryId ^= deliveryId;
```

Comment:
This bitwise operator is a really unique solution- and not one that would be easy to come to